src/main/java/seedu/exercise/model/exercise/PropertyManager.java

```diff
@@ -0,0 +1,157 @@
+package seedu.exercise.model.exercise;
+
+import static seedu.exercise.logic.parser.CliSyntax.setPrefixesSet;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.exercise.logic.parser.Prefix;
+
+/**
+ * Helps to keep track of all the existing prefixes and full names for both default and custom properties.
+ * It also helps to keep track of all the existing custom properties that have been defined by the user.
+ */
+public class PropertyManager {
+    private static final List<CustomProperty> CUSTOM_PROPERTIES = new ArrayList<>();
+    private final Set<Prefix> prefixes = new HashSet<>();
+    private final Set<String> fullNames = new HashSet<>();
+
+    /**
+     * Initialises an instance of {@code PropertyManager} object.
+     *
+     * @param prefixes         the set of prefixes to be added to the {@code PropertyManager}
+     * @param fullNames        the set of full names to be added to the {@code PropertyManager}
+     * @param customProperties the set of custom properties to be added to the {@code PropertyManager}
```

Comment:
There is a sudden change in spacing for documentation when you have the `@param or @return` annotations. Should we all follow this? @garylyp @kwekke @jietung. This is the behaviour of intelliJ's auto indentation right?

Reply:
Yeah it is intellij's auto indentation. Can find it under Settings>Editor>CodeStyle>Java>JavaDoc
Under alignment, we can opt to align parameter descriptions.
I don't mind following this format; makes it neater.

Reply:
Alright. Let's all follow this alignment if you have the annotations for docs @garylyp @jietung

Reply:
Noted.

Reply:
ok